src/services/authentication/strategies/TSPStrategy.js

```diff
@@ -50,36 +52,64 @@ class TSPStrategy extends AuthenticationBaseStrategy {
 			logger.error('TSP ticket not valid.', err);
 			throw new NotAuthenticated('TSP ticket is not valid.');
 		}
-		const { app } = this;
-		let user = await this.findUser(app, decryptedTicket);
-		if (!user) {
-			// User might have been created since the last sync
-			await app.service('sync').find({
-				query: {
-					target: SYNCER_TARGET,
-					config: {
-						schoolIdentifier: decryptedTicket.ptscSchuleNummer,
-					},
-				},
-			});
-			user = await this.findUser(app, decryptedTicket);
-			if (!user) {
-				// User really does not exist
-				throw new NotAuthenticated('User does not exist');
-			}
+
+		const now = Date.now() / 1000;
+		if (decryptedTicket.iat > now || now > decryptedTicket.exp) {
+			throw new NotAuthenticated('TSP token expired.');
 		}
 
+		// translate TSP roles into SC roles
+		let roles;
 		if (decryptedTicket.ptscListRolle && typeof decryptedTicket.ptscListRolle === 'string') {
-			const roles = decryptedTicket.ptscListRolle.split(',').map((tspRole) => ({
+			roles = decryptedTicket.ptscListRolle.split(',').map((tspRole) => ({
 				schueler: 'student',
 				lehrer: 'teacher',
 				admin: 'administrator',
 			}[tspRole.toLowerCase()])).filter((r) => r);
-			if (roles.length > 0) {
-				await app.service('users').patch(user._id, { roles });
-			}
 		}
 
+		const { app } = this;
+		let user = await this.findUser(app, decryptedTicket);
+		if (!user) {
+			// User has been created since the last sync, so we'll create him from the ticket info
+			const sourceOptions = {
+				[SOURCE_ID_ATTRIBUTE]: decryptedTicket.authUID,
+			};
+			const school = await findSchool(app, decryptedTicket.ptscSchuleNummer);
```

Comment:
findSchool return null if nothing found, without message. It should handle in helper, or on this place.

Reply:
yes, handling in the calling code is expected (new schools won't be found in the database before they are created)